api/src/main/java/org/pmiops/workbench/survey/NewUserSatisfactionSurveyServiceImpl.java

```diff
@@ -36,4 +68,54 @@ private Instant eligibilityWindowStart(DbUser user) {
   public Instant eligibilityWindowEnd(DbUser user) {
     return eligibilityWindowStart(user).plus(61, ChronoUnit.DAYS);
   }
+
+  private boolean oneTimeCodeValid(DbOneTimeCode oneTimeCode) {
+    return oneTimeCode.getUsedTime() == null
+        && oneTimeCode.getPurpose() == Purpose.NEW_USER_SATISFACTION_SURVEY
+        && eligibleToTakeSurvey(oneTimeCode.getUser());
+  }
+
+  @Override
+  public boolean oneTimeCodeStringValid(String oneTimeCode) {
+    return oneTimeCodeDao.findByStringId(oneTimeCode).map(this::oneTimeCodeValid).orElse(false);
+  }
+
+  @Override
+  @Transactional
+  public void createNewUserSatisfactionSurveyWithOneTimeCode(
+      CreateNewUserSatisfactionSurvey createNewUserSatisfactionSurvey, String oneTimeCode)
+      throws InvalidOneTimeCodeException {
+    DbOneTimeCode dbOneTimeCode =
+        oneTimeCodeDao.findByStringId(oneTimeCode).orElseThrow(InvalidOneTimeCodeException::new);
+    if (!oneTimeCodeValid(dbOneTimeCode)) {
+      throw new InvalidOneTimeCodeException();
+    }
+    newUserSatisfactionSurveyDao.save(
+        newUserSatisfactionSurveyMapper.toDbNewUserSatisfactionSurvey(
+            createNewUserSatisfactionSurvey, dbOneTimeCode.getUser()));
+    dbOneTimeCode = oneTimeCodeDao.save(dbOneTimeCode.setUsedTime(Timestamp.from(clock.instant())));
+  }
+
+  @Override
+  public void emailNewUserSatisfactionSurveyLinks() {
+    for (DbUser user : userDao.findAll()) {
+      final boolean previouslyEmailedSurvey =
+          user.getOneTimeCodes().stream()
+              .anyMatch((code) -> code.getPurpose() == Purpose.NEW_USER_SATISFACTION_SURVEY);
+      if (!previouslyEmailedSurvey && eligibleToTakeSurvey(user)) {
+        DbOneTimeCode dbOneTimeCode =
+            oneTimeCodeDao.save(
+                new DbOneTimeCode().setUser(user).setPurpose(Purpose.NEW_USER_SATISFACTION_SURVEY));
+        final String surveyLink =
+            String.format(
+                "%s?surveyCode=%s",
+                workbenchConfigProvider.get().server.uiBaseUrl, dbOneTimeCode.getId().toString());
+        try {
+          mailService.sendNewUserSatisfactionSurveyEmail(user, surveyLink);
```

Comment:
Need to check with support/Moira: We did not store those emails if we are just senders. If we want RW stores those emails, we need to be CCed.  I got this request from supports for another email, then I end up with supporting CC: https://github.com/all-of-us/workbench/pull/5599